services/asymkey/commit.go

```diff
@@ -391,16 +407,49 @@ func ParseCommitWithSSHSignature(ctx context.Context, c *git.Commit, committer *
 
 		for _, k := range keys {
 			if k.Verified && activated {
-				commitVerification := verifySSHCommitVerification(c.Signature.Signature, c.Signature.Payload, k, committer, committer, c.Committer.Email)
+				commitVerification := verifySSHCommitVerification(c.Signature.Signature, c.Signature.Payload, k, committerUser, committerUser, c.Committer.Email)
 				if commitVerification != nil {
 					return commitVerification
 				}
 			}
 		}
 	}
 
+	// Try the pre-set trusted keys (for key-rotation purpose)
+	for _, k := range setting.Repository.Signing.TrustedSSHKeys {
+		// FIXME: why here uses "commiterUser" as "signerUser" but below don't? why here uses "c.Committer.Email" but below uses "gpgSettings.Email"?
+		signerUser := committerUser
+		commitVerification := verifySSHCommitVerificationByInstanceKey(c, committerUser, signerUser, c.Committer.Email, k)
+		if commitVerification != nil && commitVerification.Verified {
+			return commitVerification
+		}
```

Comment:
FIXME: about "signerUser"

Actually I have one more question: is it fine to use "TrustedSSHKeys" like this? For example:

1. Set SigningKey=K1, SigningEmail=Email1
2. Then rotate, set SigningKey=K2, SigningEmail=Email2, add K1 to TrustedSSHKeys
3. Now, when use K1 to verify, it knows nothing about Email1, so some "verification trust models" would always fail?